### PR TITLE
Add in check for local preview for generated maps 

### LINKED ISSUE #1881: Use local preview for generated maps
I would like to display the generated map preview for the host when generating maps. This will allow host to easily regenerate if the map is not to their liking rather than having to go all the way into the lobby.

This should be implemented as checking for the locally created preview and if it is not there then using the image stand in.

### PR DESCRIPTION
**Pull request by Sheikah45**
Fixes #1881

### CODE REVIEW

File: src/main/java/com/faforever/client/map/MapService.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/map/MapService.java:306]: Just that you know I think having the random icon was a feature and was not done because guys that wrote it did not know how to fetch the preview...

But since you are the maintainer of map gen I leave it to u, just think about it that maybe having the mystery about how the map looks until you join might be interesting.

File: src/main/java/com/faforever/client/map/MapService.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/map/MapService.java]: Why do we have an AWT image only to convert it to an java fx image later, makes little sense to me pls try do this the java fx way...

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/map/MapService.java
Comment (Sheikah45) [reply src/main/java/com/faforever/client/map/MapService.java]: yeah i was just lazy and didnt change the import settings but will do so

File: src/main/java/com/faforever/client/map/MapService.java
Comment (Sheikah45) [reply src/main/java/com/faforever/client/map/MapService.java:306]: yeah I was just considering whether it should show the preview since we also added the options but am honestly not sure about this yet just implemented to see. This also will only get it for the host before they create the game

File: src/main/java/com/faforever/client/map/MapService.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/map/MapService.java]: Can u not do new Image(....getURL())

Comment (Sheikah45) [reply src/main/java/com/faforever/client/map/MapService.java]: There is no url for this image as this is getting the image generated on the local filesystem

Comment (Brutus5000) [reply src/main/java/com/faforever/client/map/MapService.java]: Hard to read. My proposal:

```
Path previewPath = forgedAlliancePreferences.getCustomMapsDirectory().resolve(mapName).resolve(mapName + "_preview.png");
return new Image(Files.newInputStream(previewPath));
```

I'm also not totally sure when the exception would be thrown? If the file does not exist yet? That shouldn't be catched via try-catch but with `Files.exists(previewPath)` instead (-> never use Exceptions for control flow).
If it's used for anything else, please add a comment.

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

**Review by Brutus5000 — APPROVED** [review]

**Review by 1-alex98 — APPROVED** [review]

**Pull request merged**
